vnext/Microsoft.ReactNative/ABIViewManager.cpp

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "ABIViewManager.h"
+
+#include <ReactUWP/Utils/ValueUtils.h>
+#include "ReactSupport.h"
+
+namespace winrt::Microsoft::ReactNative::Bridge {
+
+ABIViewManager::ABIViewManager(
+    const std::shared_ptr<react::uwp::IReactInstance> &reactInstance,
+    const winrt::Microsoft::ReactNative::Bridge::IViewManager &viewManager)
+    : Super(reactInstance) {
+  if (nullptr == viewManager) {
+    throw winrt::hresult_null_argument(L"viewManager");
+  }
```

Comment:
Throwing in constructor is generally bad. It is better to crash here. std::terminate.

Reply:
ideally we want to crater here and get watson dumps instead

---
In reply to: [338220040](https://github.com/microsoft/react-native-windows/pull/3490#discussion_r338220040) [](ancestors = 338220040)

Reply:
In Office we usually usually cause SEH exception and the unhandled exception crashes the app and sends Watson dump.
I have added the Crash.h header in Microsoft.ReactNative project. We can replace the std::terminate there with the SEH exception throwing code like we do in Office: 
*((int*) 0) = 1;


---
In reply to: [338277180](https://github.com/microsoft/react-native-windows/pull/3490#discussion_r338277180) [](ancestors = 338277180,338220040)

Reply:
Note, that react-native already uses std::terminate extensively in their code.
Office at least catches the std::terminates and handles them appropriately (including attempting doc recovery).  -- This may be a gap in the template code, that it should do some kind of hookup to ensure good error reporting for std::terminate.